lib/vicadmin/validate.go

```diff
@@ -191,3 +202,55 @@ func (v *Validator) QueryDatastore(ctx context.Context, vch *config.VirtualConta
 			</div>`, v.StorageRemaining, ds.Name, float64(ds.Summary.FreeSpace)/(1<<30)))
 	}
 }
+
+func (v *Validator) QueryVCHStatus(vch *config.VirtualContainerHostConfigSpec) {
+	defer trace.End(trace.Begin(""))
+	v.VCHIssues = template.HTML("")
+	v.VCHStatus = GoodStatus
+
+	procs := []string { "vic-init" }
+
+	// Extract required components from vchConfig
+	// Only report on components with Restart set to true
+	for _, sess := range vch.ExecutorConfig.Sessions {
+		if !sess.Restart {
+			continue
+		}
+		parsedcmd := basename.FindStringSubmatch(sess.Cmd.Path)
+		procs = append(procs, parsedcmd[1])
+	}
+	log.Infof("Processes to check: %+v", procs)
+
+	for _, proc := range procs {
+		log.Infof("Checking status of %s", proc)
+		pid, err := ioutil.ReadFile(fmt.Sprintf("%s/%s.pid", tether.PIDFileDir, proc))
+		if err != nil {
+			v.VCHIssues = template.HTML(fmt.Sprintf("%s<span class=\"error-message\">Process %s not running</span>\n",
+				v.VCHIssues, proc))
+			log.Errorf("Process %s not running: %s", proc, err)
+			continue
+		}
+
+		status, err := ioutil.ReadFile(fmt.Sprintf("/proc/%s/stat", string(pid)))
+		if err != nil {
+			v.VCHIssues = template.HTML(fmt.Sprintf("%s<span class=\"error-message\">Unable to stat process %s</span>\n",
+				v.VCHIssues, err))
+			continue
+		}
+
+		fields := strings.Split(string(status), " ")
+		// Field 3 is the current state as reported by the kernel
+		switch fields[2][0] {
+		// We're good
+		case 'R', 'S', 'D': break
+		// Process has been killed, is dying, or a zombie
+		case 'K', 'X', 'x', 'Z':
+			v.VCHIssues = template.HTML(fmt.Sprintf("%s<span class=\"error-message\">Process %s has failed</span>\n",
```

Comment:
If this is going to be displayed to the user, I would use the name of the service whose process if failing.